### PR TITLE
Adding run-parts for userdel

### LINKED ISSUE #79: Consider adding an option to remove user journals on userdel
Maybe systemd should be handling this itself but just in case,
would it be possible to have userdel have an option to remove /var/log/journal/<machine_id>/user-1001*.journal when the user with id 1001 is deleted?
This would prevent the next user with the same id from reading the old user's data.
For example, when doing userdel testuser2 and useradd testuser3 since they will have the same user id.
userdel already offers to remove mail spool data through an option.

### PR DESCRIPTION
**Pull request by edneville**
This should enable additional user tidy-up when running userdel.

Happy to repeat elsewhere if this is acceptable.

Closes #79.

### CODE REVIEW

**Comment by hallyn** [comment]
Is /etc the right place for this?  Would /usr/share or something be better?  Is the intent for distro-specific packages to ship some hooks, or only for admins to be able to define them?  (/etc seems right in the latter case)  Do you have a specific use case for this?

**Comment by edneville** [comment]
/etc feels like the right place for it, sort of akin to /etc/skel which is user setup time. It seems to me the sort of place that an admin would look.

At my day job we have some "cloud" computers, which unfortunately allow people to create users via the web portal. I'd like to capture as much of that action at useradd time, collect the tty, time, source and send that as a mail so the originator can be contacted. Clouds probably have specific end points that portal work comes from which would make life easier also.

Other ways may also exist to do the above, but if I can think of this, there's probably lots of things I can't think of also. Audit does seem to be a good starting point. Log aggregation may also benefit from this if particular strings are added in syslog that show were the additions and deletions come from in a way that is easy to manage.

File: lib/run_part.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+		if (S_ISREG (sb.st_mode)) {
+			execute_result = run_part (s, name, system);
+		}
+
+		free (s);
+		free (namelist[n]);
+
+		if (execute_result!=0) {
+			break;
```
Comment (hallyn) [inline lib/run_part.c]: In this case you'll leak namelist[n+1..scanlist]

Comment (edneville) [reply lib/run_part.c]: Good point, this should be fixed now.

**Comment by edneville** [comment]
@hallyn, I've updated the pull request with garbage collection. Thanks to spotting that.

**Comment by hallyn** [comment]
Hi


Please squash these all into one commit.

The api for talking to the scripts feels a little odd to me.  Are you already using this?  In particular, I mean passing "USERDEL=username" to the script, instead of something like ACTION=userdel SUBJECT=username or something.  If you're already using it, we can go with how it is, and maybe my aesthetic sense is wrong here?

In the run_parts loop, you only execute if the file is S_ISREG, but what about a symlink?

Also, if the script fails, perhaps you should log which script fails?

Other than that, I'm happy to merge this.

**Comment by edneville** [comment]
Hi @hallyn 

I can't squash the commits in my branch (as far as I know) without deleting the branch and creating a new pull request, which then might clutter the original issue ticket. I /think/ you can squash the merge though which should collapse the commits into one commit.

This isn't in production use yet, only test, so changing the environment variable names isn't a problem for me.

Thanks for the comments, looking forward to getting this one crossed off.

**Comment by hallyn** [comment]
Hi,

try this:

git checkout 79_userdel # or whatever your current local branch name for this pr is
git checkout -b 79_userdel.rebase
git rebase -i HEAD~6
# in the editor, replace each 'pick' - except the first one\ - with 's' for squash.
git push --force edneville HEAD:79_userdel  # this is assuming your local name for your own remote is 'edneville'

I'll do it if you prefer, but you most certainly should be able to :)

**Comment by edneville** [comment]
Hi Serge,

Thanks for that. I've updated my branch, I didn't think a pushed branch 
could be re-written, every day, something new!

Ed

**Comment by edneville** [comment]
Hi @hallyn, is this waiting on anything from me or could this be merged now?

**Comment by hallyn** [comment]
Oh, sorry - my implicit mental filter mistakenly thought there was more to discuss.

I'll take one last brief look and merge, thanks!

**Comment by hallyn** [comment]
Just ONE thing - in some of the run_parts error cases you do not free namelist itself.  We'll be exiting immediately so it's hard to care too much, but one never knows how the code will evolve.

**Comment by edneville** [comment]
@hallyn, yes, I see your point. I've updated the PR to free.

**Comment by hallyn** [comment]
Great, thanks - there's only one other little detail - in the last hunk where you add 'perror', you do the perror after the frees.  That will change the errno value so perror won't report stat's errno.

**Comment by hallyn** [comment]
I'll go ahead and make the change and merge.

**Pull request merged**